---
title: "DeSeq_Analysis_D1&4vsD2&5"
author: "Mei Yu"
date: "2022-03-22"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
options(stringsAsFactors = FALSE)
options(java.parameters = "-Xmx8000m")
set.seed(142)
```

```{r load packages}
library(DESeq2)
library(tximport)
library(biomaRt)
library(ggplot2)
library(ggrepel)
library(clusterProfiler)
library(pheatmap)
library(genefilter)
library(org.Mm.eg.db)
library(gplots)
library(EnhancedVolcano)
library(enrichplot)
library(ReactomePA)
library(readxl)
library(Rtsne)
library(tidyverse)
```

```{r load data}
setwd("/volumes/Lychee/Mei_Backup/Projects/MM_Madison/RNASeq/")
countdata.total <- as.data.frame(readxl::read_xlsx("Gene_Counts_WT_D1_D2_D4_D5.xlsx"))
rownames(countdata.total) <- countdata.total$Symbol
countdata.total$Symbol <- NULL

coldata.total <- read.csv("/Volumes/Lychee/Mei_Backup/Projects/MM_Madison/RNASeq/Sample_Metadata.csv", header = T)

rownames(coldata.total) <- coldata.total$sample_name
summary(rownames(coldata.total) == colnames(countdata.total))
```

```{r limma transform}
dds.total <- DESeqDataSetFromMatrix(countData = countdata.total, colData = coldata.total, ~sample_group+tissue, tidy = FALSE)
dds.total <- dds.total[rowSums(DESeq2::counts(dds.total))>1, ]
library(limma)
rld.total <- rlog(dds.total)
dds.total <- DESeq(dds.total)
```

```{r Deseq}
## Deseq analysis D14 vs D25 ##
res_D14vsD25 <- results(dds.total, contrast = c("sample_group", "D1_4", "D2_5"))
res_shrink_D14vsD25 <- lfcShrink(dds.total, 
                                type = "ashr",
                                contrast = c("sample_group", "D1_4", "D2_5"))
DESeq2::plotMA(res_D14vsD25, ylim = c(-10,10))
DESeq2::plotMA(res_shrink_D14vsD25, ylim = c(-10,10))
sig_res_D14vsD25 <- as.data.frame(res_shrink_D14vsD25)[!is.na(res_shrink_D14vsD25$padj) & res_shrink_D14vsD25$padj < 0.05,]
sig_res_D14vsD25_2 <- assay(rld.total)[rownames(res_shrink_D14vsD25)[!is.na(res_shrink_D14vsD25$padj) & res_shrink_D14vsD25$padj < 0.05],]
mat_D14vsD25 <- sig_res_D14vsD25_2[, c(4:19)]
p <- pheatmap(mat_D14vsD25, 
         color = colorRampPalette(c("blue", "white", "red"))(50),
         cluster_cols = T, cluster_rows = TRUE, 
         clustering_method = "ward.D2", scale = "row", 
         border_color = NA,
         show_rownames = F,
         main = "Sig Diff Genes D1&D4 vs D2&D5",
         labels_col = coldata.total$sample_name[c(4:19)], 
         annotation_col = data.frame(row.names = rownames(coldata.total), 
                                     group = coldata.total$sample_group, tissue = coldata.total$tissue))
#ggsave("Heatmap_Sig_Diff_Genes_D1&4_VS_D2&5.pdf", plot = p, dpi = 300)
```

```{r Volcano Plot}
## volcano plot ##
EnhancedVolcano(res_shrink_D14vsD25,
                lab = rownames(res_shrink_D14vsD25),
                x = 'log2FoldChange',
                y = 'padj',
                xlim = c(-6, 6),
                ylim = c(0, -log10(10e-130)),
                axisLabSize = 10,
                pCutoff = 0.05,
                FCcutoff = 1,
                cutoffLineType = 'twodash',
                cutoffLineWidth = 0.8,
                pointSize = 2.0,
                labSize = 4.0,
                colAlpha = 1,
                legendLabels =c('NS','Log (base 2) fold-change','P value',
                                'P value & Log (base 2) fold-change'),
                legendPosition = 'bottom',
                legendLabSize = 9,
                legendIconSize = 3.0, 
                title = "D1&D4_vs_D2&D5")

# Volcano plot with Myc highlighted #
library(ggrepel)
View(as.data.frame(res_shrink_D14vsD25))
highlight <- c("Myc")
result <- as.data.frame(res_shrink_D14vsD25) %>% mutate(highlight = rownames(as.data.frame(res_shrink_D14vsD25)) %in% highlight)
result$label <- NULL
result$label[result$highlight] <- rownames(result)[result$highlight]
summary(as.factor(result$label))
myc <- result[result$highlight,]
sig_result <- result[!is.na(result$padj) & result$pvalue <= 0.05 & abs(result$log2FoldChange) >= 1,]

result <- result %>% arrange(highlight)
keyvals <- ifelse(abs(result$log2FoldChange) >= 1 & result$pvalue <= 0.05, 'darkred',  ifelse(rownames(result) == 'Myc', 'royalblue','gray'))
keyvals[is.na(keyvals)] <- 'gray'
names(keyvals)[keyvals == 'darkred'] <- 'significant'
names(keyvals)[keyvals == 'gray'] <- 'not significant'
names(keyvals)[keyvals == 'royalblue'] <- 'Myc'
summary(as.factor(keyvals))
p <- EnhancedVolcano(result,
                lab = rownames(result),
                x = 'log2FoldChange',
                y = 'pvalue',
                selectLab = rownames(result)[which(names(keyvals) %in% c('Myc'))],
                xlim = c(-12.5, 25),
                ylim = c(0, -log10(10e-160)),
                axisLabSize = 10,
                pCutoff = 0.05,
                FCcutoff = 1,
                cutoffLineType = 'twodash',
                cutoffLineWidth = 0.8,
                pointSize = c(ifelse(rownames(result) == 'Myc', 2, 1)),
                labSize = 4.0,
                labCol = "royalblue",
                colAlpha = 1,
                legendPosition = 'right',
                legendLabSize = 9,
                legendIconSize = 3.0, 
                drawConnectors = T,
                widthConnectors = 0.5,
                title = "VQ-D1/D4 vs VQ-D2/D5 RNA-Seq Volcano Plot",
                subtitle = "genes up-regulated 1565 \ngenes down-regulated 1480",
                colCustom = keyvals,
                legendLabels = NULL)
p$data <- p$data %>% arrange(-highlight)
p

D14vsD25.volcano <- ggplot(result, aes(label = label, color = "red")) +
  geom_point(aes(x = log2FoldChange, y = -log10(pvalue)), col = "black", size = 2) +
  geom_point(data = myc, aes(x = log2FoldChange, y = -log10(pvalue)), colour = "red", size = 2) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "black") +
  geom_vline(xintercept = c(1, -1), linetype = "dashed", col = "black") +
  theme_minimal() + theme(legend.position = "none") +
  theme(text = element_text(size = 20)) +
  theme(axis.line = element_line(colour = "black", size = 1, linetype = "solid")) +
  xlim(-12.5, 25) + ylim(-1, 200) +
  geom_text_repel(aes(x = log2FoldChange, y = -log10(pvalue)), max.overlaps = Inf, size = 5)+
  ggtitle("VQ-D1/D4 vs VQ-D2/D5 RNA-Seq Volcano Plot") +
  labs(caption = "genes up-regulated 1565 \n genes down-regulated 1480") +
  theme(plot.title=element_text(size=16))
D14vsD25.volcano
#ggsave("D1&D4_vs_D2&D5_RNASeq_volcano_plot_010523.pdf", D14vsD25.volcano, dpi = 300, width = 7, height = 7, units = "in")

```

```{r GSEA}
#genelist
D14vsD25_lfc_list <- res_shrink_D14vsD25$log2FoldChange
names(D14vsD25_lfc_list) <- rownames(res_shrink_D14vsD25)
D14vsD25_lfc_list <- D14vsD25_lfc_list[order(D14vsD25_lfc_list, decreasing = TRUE)]
names(D14vsD25_lfc_list) <- mapIds(org.Mm.eg.db, 
                                    keys = names(D14vsD25_lfc_list), 
                                    keytype = "SYMBOL", column = "ENTREZID")
#gsea kegg
D14vsD25_gsea_kegg <- gseKEGG(geneList = D14vsD25_lfc_list, 
                             organism = "mmu", pvalueCutoff = 1, nPerm = 10000)
D14vsD25_gsea_kegg <- setReadable(D14vsD25_gsea_kegg, 
                                 org.Mm.eg.db, keyType = "ENTREZID")
sig_D14vsD25_gsea_kegg <- D14vsD25_gsea_kegg[which(D14vsD25_gsea_kegg@result$p.adjust<0.25), ]
View(as.data.frame(sig_D14vsD25_gsea_kegg))
#gsea reactome
D14vsD25_gsea_reactome <- gsePathway(geneList = D14vsD25_lfc_list, 
                                    organism = "mouse", pvalueCutoff = 1, nPerm = 10000)
D14vsD25_gsea_reactome <- setReadable(D14vsD25_gsea_reactome, 
                                     org.Mm.eg.db, keyType = "ENTREZID")
sig_D14vsD25_gsea_reactome <- D14vsD25_gsea_reactome[which(D14vsD25_gsea_reactome@result$p.adjust<0.25),]
View(sig_D14vsD25_gsea_reactome)
#gsea Go
D14vsD25_gsea_GO <- gseGO(geneList = D14vsD25_lfc_list,
                         OrgDb = org.Mm.eg.db, pvalueCutoff = 1, nPerm = 10000)
D14vsD25_gsea_GO <- setReadable(D14vsD25_gsea_GO, 
                               org.Mm.eg.db, keyType = "ENTREZID")
sig_D14vsD25_gsea_GO <- D14vsD25_gsea_GO[which(D14vsD25_gsea_GO@result$p.adjust<0.25),]
View(sig_D14vsD25_gsea_GO)

## GSEA DATABASE C6 oncogenic signature gene sets ##
library(msigdbr)
m_c6g <- msigdbr(species = "Mus musculus", category = "C6") %>%
  dplyr::select(gs_name, entrez_gene)
head(m_c6g)
D14vsD25_gsea_c6 <- GSEA(D14vsD25_lfc_list, TERM2GENE = m_c6g, pvalueCutoff = 1)
D14vsD25_gsea_c6 <- setReadable(D14vsD25_gsea_c6, org.Mm.eg.db, keyType = "ENTREZID")
sig_D14vsD25_gsea_c6 <- D14vsD25_gsea_c6[which(D14vsD25_gsea_c6@result$p.adjust<0.25),]

## GSEA Hallmark##
Hallmark <- msigdbr(species = "Mus musculus", category = "H") %>%
  dplyr::select(gs_name, entrez_gene)
head(Hallmark)

D14vsD25_gsea_H <- GSEA(D14vsD25_lfc_list, TERM2GENE = Hallmark, pvalueCutoff = 1)
D14vsD25_gsea_H <- setReadable(D14vsD25_gsea_H, org.Mm.eg.db, keyType = "ENTREZID")
View(as.data.frame(D14vsD25_gsea_H))
sig_D14vsD25_gsea_H <- D14vsD25_gsea_H[which(D14vsD25_gsea_H@result$p.adjust<0.25),]

```

```{r GSEA UAM70}
UAMS70_Genesets <- read.gmt("Genesets/UAMS70_Gene_Signature_MGI.gmt")
UAMS70_Genesets$convert <- mapIds(org.Mm.eg.db, keys = UAMS70_Genesets$gene,
                               column = "ENTREZID", keytype = "SYMBOL")
UAMS70_Genesets <- UAMS70_Genesets %>% select(term, convert) %>% rename(gene = convert)
UAMS70_UP <- UAMS70_Genesets %>% filter(term == "UAMS70_UP_GENE" & !is.na(gene)) %>% mutate(term = "UAMS70_UP_GENE") 
UAMS70_DOWN <- UAMS70_Genesets %>% filter(term == "UAMS70_DOWN_GENE" & !is.na(gene) & gene != "NULL") %>% mutate(term = "UAMS70_DOWN_GENE")
UAMS70.signature <- rbind(UAMS70_UP, UAMS70_DOWN)

D14vsD25_gsea_UAMS70 <- GSEA(D14vsD25_lfc_list, TERM2GENE = UAMS70.signature,
                          pvalueCutoff = 1)
D14vsD25_gsea_UAMS70 <- setReadable(D14vsD25_gsea_UAMS70,
                                          org.Mm.eg.db, keyType = "ENTREZID")
View(as.data.frame(D14vsD25_gsea_UAMS70))
```

```{r EMC-92 analysis}
## LOAD THE TABLE OF EMC92
## Some Symbols were not recognized as HGNC symbols, So symbols in the supplemental table were firstly converted to HGNC.symbol on https://www.syngoportal.org/convert#
library(xlsx)
EMC92_Genesets <- read.xlsx("Genesets/EMC_92_Sup.xlsx", sheetName = "Table 1")
EMC92_POS <- EMC92_Genesets %>% filter(Weighting.coefficient..beta. > 0) %>% arrange(desc(Weighting.coefficient..beta.)) %>% mutate(term = "EMC_92_POS_SIGNATURE") %>% select(term, HGNC.symbol)
EMC92_POS$rank <- 1:nrow(EMC92_POS)
EMC92_NEG <- EMC92_Genesets %>% filter(Weighting.coefficient..beta. < 0) %>% arrange(Weighting.coefficient..beta.) %>% mutate(term = "EMC_92_NEG_SIGNATURE") %>% select(term, HGNC.symbol)
EMC92_NEG$rank <- 1:nrow(EMC92_NEG)
## convert the HGNC.symbol to Mouse orthologues
library(biomaRt)
human = useMart("ensembl", dataset = "hsapiens_gene_ensembl", host = "dec2021.archive.ensembl.org")
mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl", host = "dec2021.archive.ensembl.org")
EMC92.POS.mus <- getLDS(attributes = c("hgnc_symbol"), 
                          filters = "hgnc_symbol", 
                          values = EMC92_POS$HGNC.symbol , 
                          mart = human, 
                          attributesL = c("mgi_symbol"), 
                          martL = mouse, uniqueRows=T)
EMC92.NEG.mus <- getLDS(attributes = c("hgnc_symbol"), 
                          filters = "hgnc_symbol", 
                          values = EMC92_NEG$HGNC.symbol , 
                          mart = human, 
                          attributesL = c("mgi_symbol"), 
                          martL = mouse, uniqueRows=T)
EMC92.POS.convert <- right_join(EMC92_POS, EMC92.POS.mus, by = "HGNC.symbol")
EMC92.NEG.convert <- right_join(EMC92_NEG, EMC92.NEG.mus, by = "HGNC.symbol")
EMC92.POS.signature <- EMC92.POS.convert %>% select(term, MGI.symbol) %>% rename(gene = MGI.symbol)
EMC92.NEG.signature <- EMC92.NEG.convert %>% select(term, MGI.symbol) %>% rename(gene = MGI.symbol)

# convert gene symbol to ENTRIZID #
EMC92.POS.signature$convert  <- mapIds(org.Mm.eg.db, keys = EMC92.POS.signature$gene,
                               column = "ENTREZID", keytype = "SYMBOL")
EMC92.POS.signature <- EMC92.POS.signature %>% select(term, convert) %>% rename(gene = convert)
EMC92.NEG.signature$convert  <- mapIds(org.Mm.eg.db, keys = EMC92.NEG.signature$gene,
                               column = "ENTREZID", keytype = "SYMBOL")
EMC92.NEG.signature <- EMC92.NEG.signature %>% select(term, convert) %>% rename(gene = convert)
EMC92.signature <- rbind(EMC92.POS.signature, EMC92.NEG.signature)

# gsea with EMC92 signature genesets #
D14vsD25_gsea_EMC92 <- GSEA(D14vsD25_lfc_list, TERM2GENE = EMC92.signature,
                          pvalueCutoff = 1)
D14vsD25_gsea_EMC92 <- setReadable(D14vsD25_gsea_EMC92,
                                          org.Mm.eg.db, keyType = "ENTREZID")
View(as.data.frame(D14vsD25_gsea_EMC92))
```


```{r Applying the EMC-92 signature }
#" The EMC-92 signature is applied by multiplying the normalized, mean-variance standardized and batch corrected expression values with the weighting factors as shown in Table S1”
# Applying the variance-stabilizing transformation
vsd.total <- vst(dds.total)
View(assay(vsd.total))
# Try two ways of analysis, with removal of tissue batch effect and without #
# without removal of tissue batch effect #
gene.expression.total <- assay(vsd.total)
## convert the HGNC.symbol to Mouse orthologues
#library(biomaRt)
#human = useMart("ensembl", dataset = "hsapiens_gene_ensembl", host = "dec2021.archive.ensembl.org")
#mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl", host = "dec2021.archive.ensembl.org")
EMC92.mus <- getLDS(attributes = c("hgnc_symbol"), 
                          filters = "hgnc_symbol", 
                          values = EMC92_Genesets$HGNC.symbol , 
                          mart = human, 
                          attributesL = c("mgi_symbol"), 
                          martL = mouse, uniqueRows=F)
EMC92_Genesets <- full_join(EMC92_Genesets, EMC92.mus, by = "HGNC.symbol")
View(EMC92_Genesets %>% arrange(Rank))
EMC92.MGI.coeff <- EMC92_Genesets %>% select(MGI.symbol, Weighting.coefficient..beta.) %>% filter(!is.na(MGI.symbol))
EMC92.list <- EMC92.MGI.coeff$MGI.symbol
gene.expression.EMC92 <- gene.expression.total[rownames(gene.expression.total) %in% EMC92.list, ]
EMC92.coeff.table <- EMC92.MGI.coeff %>% filter(MGI.symbol %in% rownames(gene.expression.EMC92))
# gene Tubb5 has duplicated number #
# save table with differnt Tubb5 coeff number ##
EMC92.coeff.1 <- EMC92.coeff.table %>% filter(MGI.symbol != "Tubb5" | (MGI.symbol == "Tubb5" & Weighting.coefficient..beta. == "0.0255")) %>% arrange(MGI.symbol)
EMC92.coeff.2 <- EMC92.coeff.table %>% filter(MGI.symbol != "Tubb5" | (MGI.symbol == "Tubb5" & Weighting.coefficient..beta. == "0.0221")) %>% arrange(MGI.symbol)
summary(EMC92.coeff.1$MGI.symbol == rownames(gene.expression.EMC92))
summary(EMC92.coeff.2$MGI.symbol == rownames(gene.expression.EMC92))

# calculste EMC92 risk score #
expression.coeff.1 <- data.frame(matrix(nrow = nrow(EMC92.coeff.1), ncol = ncol(gene.expression.EMC92)))
for (i in 1: ncol(gene.expression.EMC92)){
  expression.coeff.1[,i] <- gene.expression.EMC92[,i] * EMC92.coeff.1$Weighting.coefficient..beta.
}
rownames(expression.coeff.1) <- rownames(gene.expression.EMC92)
colnames(expression.coeff.1) <- colnames(gene.expression.EMC92)

expression.coeff.2 <- data.frame(matrix(nrow = nrow(EMC92.coeff.2), ncol = ncol(gene.expression.EMC92)))
for (i in 1: ncol(gene.expression.EMC92)){
  expression.coeff.2[,i] <- gene.expression.EMC92[,i] * EMC92.coeff.2$Weighting.coefficient..beta.
}
rownames(expression.coeff.2) <- rownames(gene.expression.EMC92)
colnames(expression.coeff.2) <- colnames(gene.expression.EMC92)

EMC92.risk.score.1 <- colSums(expression.coeff.1)
View(as.data.frame(EMC92.risk.score.1))
EMC92.risk.score.2 <- colSums(expression.coeff.2)
View(coldata.total)
summary.EMC92.risk.score <- cbind(coldata.total, EMC92.risk.score.1, EMC92.risk.score.2)
#write.csv(summary.EMC92.risk.score, file = "/Volumes/Lychee/Mei_Backup/Projects/MM_Madison/RNASeq/Results/EMC92_risk_core.csv")

# plot the risk score #
summary.EMC92.risk.score$sample_group <- as.factor(summary.EMC92.risk.score$sample_group)
sample.sub.group <- c("WT", "WT", "WT", "D1", "D1", "D2", "D2", "D4", "D4", "D5", "D5", "D4", "D4", "D1", "D1", "D2", "D2", "D5", "D5")
summary.EMC92.risk.score$sample_sub_group <- sample.sub.group
summary.EMC92.risk.score$sample_sub_group <- as.factor(summary.EMC92.risk.score$sample_sub_group)

#Plot the data#
# scatter plot #
p <- ggplot(summary.EMC92.risk.score, aes(x = sample_group, y = EMC92.risk.score.1, shape=tissue, fill = sample_sub_group)) + ggtitle("EMC92 risk score (Tubb5 = 0.0255)") + scale_x_discrete(limits=c("WT", "D2_5", "D1_4", "VK12653")) + geom_point(size = 3) + theme(text = element_text(size = 15)) + scale_shape_manual(values = c(21, 23, 22, 24)) + xlab ("Group") + ylab("EMC-92 Risk Score") + theme_bw() + scale_fill_hue(labels = c("D1", "D2", "D4", "D5", "VK12653", "WT"))
p
#ggsave("ScatterPlot_of_EMC92_Risk_Score_1.pdf", p, dpi = 300, height = 5, width = 5, units = "in")

#DotPlot#
p <- ggplot(summary.EMC92.risk.score, aes(x = sample_group, y = EMC92.risk.score.1, fill = sample_sub_group)) + ggtitle("EMC92 risk score (Tubb5 = 0.0255)") + scale_x_discrete(limits=c("WT", "D2_5", "D1_4", "VK12653")) + geom_dotplot(binaxis='y', stackdir='center') + theme(text = element_text(size = 15)) + xlab ("Group") + ylab("EMC-92 Risk Score") + theme_bw()
p
#ggsave("DotPlot_of_EMC92_Risk_Score_1.pdf", p, dpi = 300, height = 5, width = 5, units = "in")



q <- ggplot(summary.EMC92.risk.score, aes(x = sample_group, y = EMC92.risk.score.2, fill = sample_sub_group)) + geom_dotplot(binaxis='y', stackdir='center') + ggtitle("EMC92 risk score (Tubb5 = 0.0221)")
q
#ggsave("Plot_of_EMC92_Risk_Score_2.pdf", q, dpi = 300, height = 3, width = 5, units = "in")

# statistics one-way ANOVA #
library(rstatix)
library(ggpubr)
# One-way ANOVA #
stat1 <- aov(EMC92.risk.score.1 ~ sample_group, data = summary.EMC92.risk.score)
summary(stat1)
stat2 <- aov(EMC92.risk.score.2 ~ sample_group, data = summary.EMC92.risk.score)
summary(stat2)
# Tukey's HSD test#
Tukey1 <- TukeyHSD(stat1, conf.level = 0.95)
Tukey2 <- TukeyHSD(stat2, conf.level = 0.95)
TukeyHSD(stat1, conf.level = 0.95)
# save the results #
sink("EMC92_risk_score_statistic_analysis.txt", append = T)
cat("One-Way ANOVA, weighting coefficient of Tubb5 = 0.0255")
cat("\n\n")
aov(EMC92.risk.score.1 ~ sample_group, data = summary.EMC92.risk.score)
cat("\n\n")
summary(stat1)
cat("\n\n")
cat("Tukey HSD test, weighting coefficient of Tubb5 = 0.0255")
cat("\n\n")
TukeyHSD(stat1, conf.level = 0.95)
cat("\n\n")
cat("One-Way ANOVA, weighting coefficient of Tubb5 = 0.0221")
cat("\n\n")
aov(EMC92.risk.score.2 ~ sample_group, data = summary.EMC92.risk.score)
cat("\n\n")
summary(stat2)
cat("\n\n")
cat("Tukey HSD test, weighting coefficient of Tubb5 = 0.0221")
cat("\n\n")
TukeyHSD(stat2, conf.level = 0.95)
sink()


#T-TEST FOR MULTIPLE GROUPS#
multiple_t_test_1 <- rstatix::t_test(data = summary.EMC92.risk.score, formula = EMC92.risk.score.1 ~ sample_group)
multiple_t_test_1
multiple_t_test_2 <- rstatix::t_test(data = summary.EMC92.risk.score, formula = EMC92.risk.score.2 ~ sample_group)
multiple_t_test_2

WriteXLS::WriteXLS(x=list(as.data.frame(multiple_t_test_1),
                          as.data.frame(multiple_t_test_2)), 
                   ExcelFileName = "EMC92_risk_score_multiple_ttest_HolmBonferroniCorrection.xlsx", 
                   SheetNames = c("Tubb5=0.0255",
                                  "Tubb5=0.0221"), 
                   AdjWidth = TRUE, FreezeRow = 1)
```
```{r Generating Waterfall Plot}
## Waterfall Plot##
D14vsD25_gsea_all <- rbind(as.data.frame(D14vsD25_gsea_H),
                           as.data.frame(D14vsD25_gsea_kegg),
                           as.data.frame(D14vsD25_gsea_reactome),
                           as.data.frame(D14vsD25_gsea_GO),
                           as.data.frame(D14vsD25_gsea_c6),
                           as.data.frame(D14vsD25_gsea_UAMS70),
                           as.data.frame(D14vsD25_gsea_EMC92))
highlight <- c("HALLMARK_E2F_TARGETS",
               "HALLMARK_MYC_TARGETS_V1",
               "HALLMARK_G2M_CHECKPOINT",
               "HALLMARK_MYC_TARGETS_V2",
               "HALLMARK_TNFA_SIGNALING_VIA_NFKB",
               "HALLMARK_INTERFERON_ALPHA_RESPONSE",
               "HALLMARK_INTERFERON_GAMMA_RESPONSE",
               "NF-kappa B signaling pathway",
               "UAMS_70_GENE_SIGNATURE",
               "EMC_92_GENE_SIGNATURE")
D14vsD25_gsea_all <- D14vsD25_gsea_all %>% mutate(highlight = D14vsD25_gsea_all$Description %in% highlight)
D14vsD25_gsea_all$label[D14vsD25_gsea_all$highlight] <- D14vsD25_gsea_all$Description[D14vsD25_gsea_all$highlight]
D14vsD25_gsea_all <- D14vsD25_gsea_all[order(D14vsD25_gsea_all$highlight, decreasing = T),]
## Generating Waterfall Plot ##
library(ggrepel)
GSEA.volcano <- ggplot(D14vsD25_gsea_all, aes(label = label, color = "red")) +
  geom_point(aes(x = NES, y = -log10(p.adjust)), col = "black", size = 2.5) +
  geom_point(data = subset(D14vsD25_gsea_all, D14vsD25_gsea_all$highlight),
             aes(x = NES, y = -log10(p.adjust)), colour = "red", size = 2.5) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", col = "grey", size = 1) +
  theme_classic() +
  theme(legend.position = "none") +
  theme(text = element_text(size = 15)) +
  theme(axis.line = element_line(colour = "black", size = 0.5, linetype = "solid")) +
  geom_text_repel(aes(x = NES, y = -log10(p.adjust)), max.overlaps = Inf,
                  size = 3) +
  ggtitle("VQ-D1/D4 vs VQ-D2/D5 GSEA Volcano Plot")
GSEA.volcano
#ggsave("GSEA_VolcanoPlot_D14vsD25_20221021.pdf", GSEA.volcano, dpi = 300, width = 7, height = 7, units = "in")

waterfall <- ggplot(D14vsD25_gsea_all, aes(label = label, colour = "red")) +
  geom_point(aes(x = NES, y = p.adjust), colour = "black") +
  geom_point(data = subset(D14vsD25_gsea_all, D14vsD25_gsea_all$highlight), 
             aes(x = NES, y = p.adjust), colour = "red") +
  geom_hline(yintercept = 0.05, linetype = "dashed", col = "grey") +
  theme_minimal() + theme(legend.position = "none") +
  geom_text_repel(aes(x = NES, y = p.adjust), max.overlaps = Inf, size = 2.5)
  
waterfall 
#ggsave("GSEA_Waterfall.pdf", waterfall, dpi = 300)
```

```{r Generating GSEA plots}
## GSEA plots ##
View(as.data.frame(D14vsD25_gsea_H))
library(grid)
title <- grobTree(textGrob("UAMS_70_Signature_Genes", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob("NES = -1.74\nFDR = 0.0037", x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
showGrob()
p <- gseaplot2(D14vsD25_gsea_UAMS70_EMC92, "UAMS_70_GENE_SIGNATURE",
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(text) +
  annotation_custom(title) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
p
ggsave("GSEAplot_UAMS_70_Gene_Signature.pdf", p, dpi = 300)

title <- grobTree(textGrob("EMC_92_Signature_Genes", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob("NES = -1.42\nFDR = 0.037", x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
q <- gseaplot2(D14vsD25_gsea_UAMS70_EMC92, "EMC_92_GENE_SIGNATURE", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(title) +
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
q
ggsave("GSEAplot_EMC_92_Gene_Signature.pdf", dpi = 300)


title <- grobTree(textGrob("HALLMARK_E2F_TARGETS", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob(bquote("NES = -2.26\nFDR = 1.68X10"^-9), x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
r <- gseaplot2(D14vsD25_gsea_H, "HALLMARK_E2F_TARGETS", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  labs(title = "HALLMARK_E2F_TARGETS") +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
r
ggsave("GSEAplot_HALLMARK_E2F_TARGETS.pdf", r, dpi = 300)

title <- grobTree(textGrob("HALLMARK_MYC_TARGETS_V1", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob(bquote("NES = -2.16\nFDR = 1.68X10"^-9), x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
s <- gseaplot2(D14vsD25_gsea_H, "HALLMARK_MYC_TARGETS_V1", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) +
  annotation_custom(title) +
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
s
ggsave("GSEAplot_HALLMARK_MYC_TARGETS_V1.pdf", dpi = 300)

title <- grobTree(textGrob("HALLMARK_MYC_TARGETS_V2", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob(bquote("NES = -2.04\nFDR = 2.41X10"^-5), x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
t <- gseaplot2(D14vsD25_gsea_H, "HALLMARK_MYC_TARGETS_V2", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(title) +
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
t
ggsave("GSEAplot_HALLMARK_MYC_TARGETS_V2.pdf", dpi = 300)


title <- grobTree(textGrob("HALLMARK_G2M_CHECKPOINT", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob(bquote("NES = -2.01\nFDR = 8.53X10"^-8), x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
u <- gseaplot2(D14vsD25_gsea_H, "HALLMARK_G2M_CHECKPOINT", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(title) +
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
u
ggsave("GSEAplot_HALLMARK_G2M_CHECKPOINT.pdf", dpi = 300)

title <- grobTree(textGrob("HALLMARK_TNFA_SIGNALING_VIA_NFKB", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob(bquote("NES = 1.80\nFDR = 1.28X10"^-4), x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
v <- gseaplot2(D14vsD25_gsea_H, "HALLMARK_TNFA_SIGNALING_VIA_NFKB", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(title) +
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
v
ggsave("HALLMARK_TNFA_SIGNALING_VIA_NFKB.pdf", v, dpi = 300,
       width = 8.10, height = 6.71, units = "in")

title <- grobTree(textGrob("KEGG NF-kappa B signaling pathway", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob(bquote("NES = 1.78\nFDR = 0.03"), x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
w <- gseaplot2(D14vsD25_gsea_kegg, "mmu04064", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(title) +
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
w
ggsave("GSEA_KEGG_NFkappaB_signaling_pathway.pdf", w, dpi = 300,
       width = 8.10, height = 6.71, units = "in")

title <- grobTree(textGrob("KEGG Proteasome", x=0.25,  y=0.97, hjust=0,
                           gp=gpar(col="black", fontsize=20, fontface="plain")))
text <- grobTree(textGrob(bquote("NES = -1.61\nFDR = 0.14"), x=0.65,  y=0.85, hjust=0,
                          gp=gpar(col="black", fontsize=20, fontface="plain")))
group1 <- grobTree(textGrob("VQ-D1/D4", x=0.18,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
group2 <- grobTree(textGrob("VQ-D2/D5", x=0.78,  y=0.32, hjust=0,
                            gp=gpar(col="black", fontsize=20, fontface="plain")))
x <- gseaplot2(D14vsD25_gsea_kegg, "mmu03050", 
               color = "black",
               ES_geom = "line",
               base_size = 20,
               rel_heights = c(1.4, 0.3, 0.8),
               subplots = 1:3) +
  theme(plot.title = element_text(size = 20, hjust = 0.5)) + 
  annotation_custom(title) +
  annotation_custom(text) +
  annotation_custom(group1) +
  annotation_custom(group2) +
  coord_fixed(0.8)
x
ggsave("GSEA_KEGG_Proteasome.pdf", dpi = 300,
       width = 8.10, height = 6.71, units = "in")
```

