# Molecular characterization stratifies VQ myeloma cells into two clusters with distinct risk signatures and drug responses

## Running Title: risk-stratification of VQ myeloma lines

## Abstract

Multiple myeloma (MM) is a cancer of malignant plasma cells in the bone marrow and extramedullary sites. We previously characterized a VQ model for human high-risk MM. Different VQ lines display distinct disease phenotypes and survivals, suggesting significant intra-model variation. Here, we use whole exome sequencing and copy number variation (CNV) analysis coupled with RNA-Seq to stratify VQ lines into corresponding clusters: Group A cells had monosomy chr5 and overexpressed genes and pathways associated with sensitivity to bortezomib (Btz) treatment in human MM patients. By contrast, Group B VQ cells carried recurrent amplification (Amp) of chromosome (chr) 3 and displayed high-risk MM features, including downregulation of Fam46c, upregulation of cancer growth pathways associated with functional high-risk MM, and expression of Amp1q and high-risk UMAS-70 and EMC-92 gene signatures. Consistently, in sharp contrast to Group A VQ cells that showed short-term response to Btz, Group B VQ cells were de novo resistant to Btz in vivo. Our study highlights Group B VQ lines as highly representative of the human MM subset with ultrahigh risk.

## Methods
- B cell receptor (BCR) Immunoglobulin heavy (IgH) chain and Immunoglobulin light (IgL) chain repertoire sequencing
- Whole Exome Sequencing (WES) and copy number variation (CNV) analysis
- RNA-Seq


## SessionInfo

```
> sessionInfo()
R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Ventura 13.2.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] shazam_1.1.2                alakazam_1.2.1              ggpubr_0.6.0               
 [4] rstatix_0.7.2               xlsx_0.6.5                  msigdbr_7.5.1              
 [7] limma_3.52.4                lubridate_1.9.2             forcats_1.0.0              
[10] stringr_1.5.0               dplyr_1.1.0                 purrr_1.0.1                
[13] readr_2.1.4                 tidyr_1.3.0                 tibble_3.2.0               
[16] tidyverse_2.0.0             Rtsne_0.16                  readxl_1.4.2               
[19] ReactomePA_1.40.0           enrichplot_1.18.3           EnhancedVolcano_1.14.0     
[22] gplots_3.1.3                org.Mm.eg.db_3.15.0         AnnotationDbi_1.60.2       
[25] genefilter_1.78.0           pheatmap_1.0.12             clusterProfiler_4.7.1.003  
[28] ggrepel_0.9.3               ggplot2_3.4.1               biomaRt_2.52.0             
[31] tximport_1.24.0             DESeq2_1.36.0               SummarizedExperiment_1.26.1
[34] Biobase_2.58.0              MatrixGenerics_1.8.1        matrixStats_0.63.0         
[37] GenomicRanges_1.48.0        GenomeInfoDb_1.34.9         IRanges_2.32.0             
[40] S4Vectors_0.36.2            BiocGenerics_0.44.0        
                 
```


